requirements.txt

```diff
@@ -53,35 +48,30 @@ flask-restful==0.3.7
 flask-sqlalchemy==2.3.2
 flask==1.1.1
 frozendict==1.2
-funcsigs==1.0.2 ; python_version < '3.3'
-functools32==3.2.3.post2 ; python_version < '3'
 furl==2.0.0
 future==0.17.1
-futures==3.1.1
 graphviz==0.10.1
-html5lib==1.0.1
 idna==2.8
-ipaddress==1.0.22 ; python_version == '2.7'
 ipydatawidgets==4.0.1
-ipykernel==5.1.1
-ipympl==0.2.1
+ipykernel==5.1.2
 ipython-genutils==0.2.0
-ipython==5.8.0
+ipython==7.7.0 ; python_version >= '3.3'
```

Comment:
how is this not a conflict?
https://github.com/aiidateam/aiida-core/blob/91715883b30409397cc034704b41908fa2e42fb6/setup.json#L45

Reply:
Honestly, I don't know. If I run
```
pipenv lock --requirements > requirements.txt_stable 
pipenv lock --dev --requirements > requirements.txt_dev
```

both `requirements.txt_stable`  and `requirements.txt_dev` end up having the line
```
ipython==7.7.0 ; python_version >= '3.3'
```

Reply:
Hm... perhaps because aiida-core is a `dev` package?

I wonder what type of dependency resolution is actually done for dev packages?